### DISCUSSION TITLE
On using latest oneAPI 2022.3

### DISCUSSION THREAD

**Discussion by ghost** [ginkgo-project/ginkgo#1203] [General]
Hello:

I am wondering why is ginko is not using the latest oneAPI 2022.* release but still dependent on the 2021.3 release? Is it because of compatibility issues with ginko's SYCL backend?

**Comment by yhmtsai** [comment]
No, Ginkgo is available on 2021.3+ which also includes the 2022.*.

**Reply by tcojean** [reply]
In fact, we recommend the latest oneAPI versions because they improved a lot. 2021.3 is the first version we know Ginkgo should be able to run with.

**Comment by ghost** [comment]
@yhmtsai and @tcojean 

Thank you for your responses. I was able to compile with oneAPI 2022.3.1; although I am running into an issue with using the [intel/llvm/](https://github.com/intel/llvm) compiler

```
CMake Error at dpcpp/CMakeLists.txt:106 (target_link_libraries):
  Target "ginkgo_dpcpp" links to:

    MKL::MKL_DPCPP

  but the target was not found.  Possible reasons include:

    * There is a typo in the target name.
    * A find_package call is missing for an IMPORTED target.
    * An ALIAS target is missing.

``` 
I do have the oneMKL and oneDPL environment variables set correctly.

**Reply by yhmtsai** [reply]
Thanks for reporting it. they seem to change their cmake name. Could you try change the dpcpp/CMakeLists.txt:106 `MKL::MKL_DPCPP` to `MKL` or `MKL::MKL`?

**Reply by ghost** [reply]
@yhmtsai : Thank you for the suggestion. I did change the `dpcpp/CMakeLists.txt:106` to `MKL::MKL` and I am able to get around the issue. I am now running into the other problems related to the latest [intel/llvm](https://github.com/intel/llvm) 

e.g., `CL/sycl.hpp file not found`, `error: no type named 'buffer_allocator' in namespace 'sycl'; did you mean 'image_allocator'?` 

These errors seem to be the internal changes within the latest compiler. I will have to update the code and work around these errors.

**Reply by yhmtsai** [reply]
I think CL/sycl.hpp may be the path issue or they forget to change the path/copy file.
I do not see any usage of buffer_allocator in ginkgo. it may be also from their internal changes.

**Reply by yhmtsai** [reply]
Could you maybe try the code in multigrid_experimental? It may resolve some issue from the namespace ambiguous issue. you still need to apply the CMake changes on your side.
